noteable_magics/datasource_postprocessing.py

```diff
@@ -158,3 +163,58 @@ def postprocess_snowflake(
         db = dsn_dict['database']
 
         dsn_dict['database'] = f'{db}/{schema}'
+
+
+@register_postprocessor('sqlite')
+def postprocess_sqlite(
+    datasource_id: str, dsn_dict: Dict[str, str], create_engine_kwargs: Dict[str, Any]
+) -> None:
+    """Expect to have either a URL provided which we should download and place into TMPDIR as the database file,
+    or no such thing implying to run in memory mode.
+    """
+
+    if 'database' in dsn_dict:
+        cur_path = dsn_dict['database']
+        if cur_path == '' or cur_path == ':memory:':
+            # Empty path is alias for :memory:, and is fine.
+            return
+
+        # If it smells like a URL, we should download, stash it into a tmpfile,
+        # and respell dsn_dict['database'] to point to that file. Any exceptions in here
+        # will spoil the datasource.
+        parsed = urlparse(dsn_dict['database'])
+
+        # (Sigh, 'mock' as scheme due to cannot cleanly pytest requests-mock http or https urls
+        #  for reasons I trust from the requests-mocks docs)
+        if parsed.scheme in ('http', 'https', 'ftp', 'mock'):
+            max_download_time = 10  # seconds. What do we want this to be?
```

Comment:
Open question here for what we should cap download time at. This figure was uh pulled from regular source of figures.

A future improvement which could then allow for somewhat higher values would be to delay bootstrapping either all or individual datasources until first or this one is actually requested, instead of at kernel startup time, but that'd take some additional restructuring best left to an individual focused PR. And prior discussion.